### PR TITLE
support TCP (and add simple in-browser tester)

### PR DESCRIPTION
**Pull request by tobowers**
This adds TCP support and a really simple in-browser tester. The readme file in /browsertest describes how to run the browser test. It passes in Chrome/FF but fails in Safari. I combined these too because TCP could have been a breaking change in the browser.

### CODE REVIEW

**Comment by tobowers** [comment]
test fixed.

File: src/community/community.spec.ts [hunk 1]
```diff
@@ -89,9 +89,27 @@ describe('Community', () => {
         throw new Error("error getting id")
       }
       await c.playTransactions(tree, [setDataTransaction("/somewhere/cool", "foo")])
-      await c.nextUpdate()
 
-      const respTip = await c.getTip(id)
+      const recursiveGetTip = ():Promise<CID> => {
```
Comment (brandonwestcott) [inline src/community/community.spec.ts]: Do we expect this pattern to be needed for other clients? If so, should we just expose a method similar to this on the `Community` class itself - something like a `getTipWithRetry` or adding a retry option to `getTip`?

Comment (tobowers) [reply src/community/community.spec.ts]: I *expect* it to be but this is the only place in the code that does it so I'd like to wait for one more use.

**Comment by tobowers** [comment]
> lgtm 👍
> 
> not for this PR, but its probably useful for `test-suite` to do a headless browser and verify this also

for sure! I tried to use Karma but couldn't get it to work so need some way to yay or nay the results of the tests.

**Pull request merged**